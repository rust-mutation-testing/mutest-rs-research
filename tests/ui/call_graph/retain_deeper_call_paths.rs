//@ print-call-graph
//@ print-targets
//@ stdout
//@ stderr: empty

fn impl_drop() {}

fn pseudo_drop() {
    impl_drop();
}

#[test]
fn test_direct() {
    pseudo_drop();
}

fn indirect_drop() {
    pseudo_drop();
}

#[test]
fn test_indirect() {
    indirect_drop();
}
