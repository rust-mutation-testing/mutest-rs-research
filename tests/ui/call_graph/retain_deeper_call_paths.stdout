
@@@ call graph @@@

entry points:

test test_direct
  -> pseudo_drop at tests/ui/call_graph/retain_deeper_call_paths.rs:8:1: 8:17 (#0)
test test_indirect
  -> indirect_drop at tests/ui/call_graph/retain_deeper_call_paths.rs:17:1: 17:19 (#0)

nested calls at distance 1:

pseudo_drop at tests/ui/call_graph/retain_deeper_call_paths.rs:8:1: 8:17 (#0)
  -> impl_drop at tests/ui/call_graph/retain_deeper_call_paths.rs:6:1: 6:15 (#0)
indirect_drop at tests/ui/call_graph/retain_deeper_call_paths.rs:17:1: 17:19 (#0)
  -> pseudo_drop at tests/ui/call_graph/retain_deeper_call_paths.rs:8:1: 8:17 (#0)

@@@ targets @@@

tests -(1)-> impl_drop at tests/ui/call_graph/retain_deeper_call_paths.rs:6:1: 6:15 (#0)
  (1) test_direct
  (2) test_indirect

tests -(0)-> pseudo_drop at tests/ui/call_graph/retain_deeper_call_paths.rs:8:1: 8:17 (#0)
  (0) test_direct
  (1) test_indirect

tests -(0)-> indirect_drop at tests/ui/call_graph/retain_deeper_call_paths.rs:17:1: 17:19 (#0)
  (0) test_indirect

targets: 3 total; 3 safe; 0 unsafe (0 tainted)
